addons/point_of_sale/models/pos_session.py

```diff
@@ -2020,8 +2020,8 @@ def get_total_discount(self):
         amount = 0
         for line in self.env['pos.order.line'].search([('order_id', 'in', self.order_ids.ids), ('discount', '>', 0)]):
             normal_price = line.qty * line.price_unit
-            normal_price = normal_price + (normal_price / 100 * line.tax_ids.amount)
-            amount += normal_price - line.price_subtotal_incl
+            tax_amount = line.price_subtotal_incl - line.price_subtotal
+            amount += (normal_price + tax_amount) - line.price_subtotal_incl
```

Comment:
```suggestion
            original_price = line.price_subtotal_incl / (1 - line.discount / 100)
            amount += original_price * line.discount / 100
```

Also, remove the line above `normal_price = line.qty * line.price_unit`. There should only be 2 lines inside the for loop.